language/vm/src/file_format.rs

```diff
@@ -278,13 +294,47 @@ pub struct FunctionHandle {
 #[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
 pub enum StructFieldInformation {
     Native,
-    Declared {
-        /// The number of fields in this type.
-        field_count: MemberCount,
-        /// The starting index for the fields of this type. `FieldDefinition`s for each type must
-        /// be consecutively stored in the `FieldDefinition` table.
-        fields: FieldDefinitionIndex,
-    },
+    Declared(Vec<FieldDefinition>),
+}
+
+//
+// Instantiations
+//
+// Instantiations point to a generic handle and its instantiation.
+// The instantiation can be partial.
+// So, for example, `S<T, W>`, `S<u8, bool>`, `S<T, u8>`, `S<X<T>, address>` are all
+// `StructInstantiation`s
+
+/// A complete or partial instantiation of a generic struct
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub struct StructDefInstantiation {
+    pub def: StructDefinitionIndex,
+    pub type_parameters: SignatureIndex,
+}
+
+/// A complete or partial instantiation of a function
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub struct FunctionInstantiation {
+    pub handle: FunctionHandleIndex,
+    pub type_parameters: SignatureIndex,
```

Comment:
Same, `type_arguments` instead of `type_parameters`?

Reply:
see above...